src/cdk/testing/protractor/protractor-element.ts

```diff
@@ -198,4 +198,15 @@ export class ProtractorElement implements TestElement {
   async isFocused(): Promise<boolean> {
     return this.element.equals(browser.driver.switchTo().activeElement());
   }
+
+  async dispatchEvent(name: string, data?: Record<string | number, any>): Promise<void> {
+    return browser.executeScript(`
+      var event = document.createEvent('Event');
```

Comment:
Can you extract this into a separate function that you pass to `executeScript`? In fact it can probably just share an implementation with the testbed version

Reply:
This is basically what `dispatchFakeEvent` from `testing/private` is doing, but I couldn't reuse the function, because it's referring to other outside functions. We could reuse the function between TestBed and Protractor, but the TestBed implementation would have to incur some unnecessary performance overhead, because the `data` literal has to be stringified and parsed for Protractor to work correctly.

Reply:
Ah ok that makes sense, even if it can't be shared can you extract it so we get type checking and nicer parameter names for the method?

Reply:
Done.